changelogs/unreleased/restore-cwd-correctly-after-each-test.yml

```diff
@@ -0,0 +1,4 @@
+---
+description: Ensure that the cwd is restored correctly after each test.
+change-type: patch
+destination-branches: [master]
```

Comment:
@wouterdb @sanderr Should I merge this fix on ISO4 and ISO3 as well? It doesn't cause issues there, but the bug still exists.